src/app/components/NeedForm/NeedForm.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import type { Need } from '../../types/types'
+import Button from '../Button/Button'
+
+type NeedFormProps = {
+  onSubmit: (newTopic: Need) => void
+  onCancel: () => void
+}
+
+const MAX_DESCRIPTION_LENGTH = 80
+
+export default function NeedForm({
+  onSubmit,
+  onCancel,
+}: NeedFormProps): JSX.Element {
+  const [text, setText] = useState('')
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const newNeed = {
+      id: Math.floor(Math.random() * 10000),
+      text,
+    }
+    onSubmit(newNeed)
+  }
+
+  function handleCancel() {
+    setText('')
+    onCancel()
+  }
+
+  const textDiff = MAX_DESCRIPTION_LENGTH - text.length
+
+  return (
+    <TopicFormContainer onSubmit={handleSubmit}>
+      <FormTitle>Add Need</FormTitle>
+      <InputWrapper>
+        <FormLabel htmlFor="text">Need</FormLabel>
+        <FormTextArea
+          name="text"
+          rows={10}
+          value={text}
+          onChange={event =>
+            event.target.value.length <= MAX_DESCRIPTION_LENGTH &&
+            setText(event.target.value)
```

Comment:
Why not use the maxlength attribute of a textarea?

Reply:
hmm good point, I didn't know that it exists.